### PR TITLE
[CWS] rework secprofile tryAutolearn

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Rework needed to handle hands of corner cases.

Also, add 44 unitary tests around it

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/security_profile/profile/manager.go [hunk 1]
```diff
@@ -681,50 +683,76 @@ func (m *SecurityProfileManager) tryAutolearn(profile *SecurityProfile, event *m
...
 		profileState = WorkloadWarmup
 	} else {
-		// have we reached the stable state time limit ?
-		lastAnomalyNano, ok := profile.lastAnomalyNano[event.GetEventType()]
-		if !ok {
-			profile.lastAnomalyNano[event.GetEventType()] = profile.loadedNano
-			lastAnomalyNano = profile.loadedNano
+		// If for the given event type we already are on StableEventType (and outside of the warmup period), just return
+		if eventState.state == StableEventType {
+			m.incrementEventFilteringStat(event.GetEventType(), StableEventType, NA)
```
Comment (Gui774ume) [inline pkg/security/security_profile/profile/manager.go:709]: When we answer `StableEventType`, we already count the event later. I think we should remove this.

Comment (spikat) [reply pkg/security/security_profile/profile/manager.go:709]: thanks

File: pkg/security/security_profile/profile/manager_test.go [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+	eventProcessPath    string                     // exec path
+	eventDnsReq         string                     // dns request name (only for eventType == DNSEventType)
+	loopUntil           time.Duration              // if not 0, will loop until the given duration is reached
+	loopIncrement       time.Duration              // if loopUntil is not 0, will increment this duration at each loop
+}
+
+func craftFakeEvent(t0 time.Time, ti *testIteration, defaultContainerID string) *model.Event {
+	event := model.NewDefaultEvent().(*model.Event)
+	event.Type = uint32(ti.eventType)
+	event.ContainerContext.CreatedAt = uint64(t0.Add(ti.containerCreatedAt).Unix()) * 1000000000
```
Comment (safchain) [inline pkg/security/security_profile/profile/manager_test.go:42]: UnixNano ?

**Review by safchain — APPROVED** [review]

File: pkg/security/security_profile/profile/manager_test.go (hunk 2 above)
Comment (spikat) [reply pkg/security/security_profile/profile/manager_test.go]: Fixed! Thank you

**Pull request merged**